zapier push fails with "Cannot find module 'supports-color'"

When running `zapier push`, I get the following error : 
```
Preparing to build and upload your app.

  Copying project to temp directory ⠃ done!
  Installing project dependencies ⠃ done!
  Applying entry point file ⠃ done!
  Building app definition.json ⠃ done!
  Validating project ⠃ done!
  Zipping project and dependencies ⠰events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: Cannot find module 'supports-color' from '/private/var/folders/dw/d_f_yf1x5zz2fm1rtkpxzmw00000gn/T/zapier-19a7602e/node_modules/debug/src'
    at /Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:48:31
    at processDirs (/Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:182:39)
    at ondir (/Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:197:13)
    at load (/Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:80:43)
    at onex (/Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:105:17)
    at /Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:26:73
    at FSReqWrap.oncomplete (fs.js:123:15)
```

Previously it was working fine ...

Output of `zapier push --debug` : 

```
Preparing to build and upload your app.

>> GET https://zapier.com/api/platform/cli/check
<< 200
<< {"success": true}

  Copying project to temp directory ⠃ done!
  Installing project dependencies ⠘

Running npm install --production command in /private/var/folders/dw/d_f_yf1x5zz2fm1rtkpxzmw00000gn/T/zapier-c0fec6e8:

⠉gofer-zapier@1.0.0 /private/var/folders/dw/d_f_yf1x5zz2fm1rtkpxzmw00000gn/T/zapier-c0fec6e8
├─┬ axios@0.17.1 
│ ├─┬ follow-redirects@1.2.6 
│ │ └─┬ debug@3.1.0 
│ │   └── ms@2.0.0 
│ └── is-buffer@1.1.6 
├─┬ graphql-request@1.4.0 
│ └─┬ cross-fetch@0.0.8 
│   ├── node-fetch@1.7.3 
│   └── whatwg-fetch@2.0.3 
├── moment@2.19.2 
└─┬ zapier-platform-core@3.3.0 
  ├── bluebird@3.5.0 
  ├── content-disposition@0.5.2 
  ├── dotenv@4.0.0 
  ├─┬ form-data@2.2.0 
  │ ├── asynckit@0.4.0 
  │ ├─┬ combined-stream@1.0.5 
  │ │ └── delayed-stream@1.0.0 
  │ └─┬ mime-types@2.1.17 
  │   └── mime-db@1.30.0 
  ├── lodash@4.17.4 
  ├─┬ node-fetch@1.7.1 
  │ ├─┬ encoding@0.1.12 
  │ │ └── iconv-lite@0.4.19 
  │ └── is-stream@1.1.0 
  └─┬ zapier-platform-schema@3.3.0 
    └── jsonschema@1.1.1 


⠃ done!
  Applying entry point file ⠃ done!
  Building app definition.json ⠃ done!
  Validating project ⠤>> POST https://zapier.com/api/platform/cli/style-check
>> {"version":"1.0.0","platformVersion":"3.3.0","beforeRequest":[],"afterResponse":[],"resources":{},"triggers":{"contractValidation":{"key":"contractValidation","noun":"Contract Validation","display":{"label":"New Contract Validation","description":"Trigger when a contract is signed by a worker."},"operation":{"perform":"$func$2$t$"}}},"searches":{},"creates":{},"searchOrCreates":{},"zip_file":"raw zip removed in logs"}
<< 200
<< {"errors": {}, "warnings": {}}

⠃ done!
  Zipping project and dependencies ⠉events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: Cannot find module 'supports-color' from '/private/var/folders/dw/d_f_yf1x5zz2fm1rtkpxzmw00000gn/T/zapier-c0fec6e8/node_modules/debug/src'
    at /Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:48:31
    at processDirs (/Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:182:39)
    at ondir (/Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:197:13)
    at load (/Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:80:43)
    at onex (/Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:105:17)
    at /Users/aymericbouzy/.nvm/versions/node/v6.10.2/lib/node_modules/zapier-platform-cli/node_modules/resolve/lib/async.js:26:73
    at FSReqWrap.oncomplete (fs.js:123:15)
```

The truth is that's not necessary for `mocha`, for example (so that information in the documentation could cause confusion), so it seems related to `jest` and I agree it's really strange.

We do have this information in https://zapier.github.io/zapier-platform-cli/#using-npm-modules:

> Note: If your package isn't being pushed correctly (IE: you get "Error: Cannot find module 'whatever'" in production), try adding the `--disable-dependency-detection` flag to `zapier push`.

Does that work for you (while keeping `jest` in devDependencies?)

Also, could you tell me what's your app id (look in `.zapierapprc`)? I'll look at the built code and try to understand if it's something with `browserify` ()

We have a babel example that doesn't need that, but babel is in its dependencies: https://github.com/zapier/zapier-platform-example-app-babel/blob/master/package.json

--- a/package.json
+++ b/package.json
@@ -33,7 +33,7 @@
     "archiver": "1.3.0",
     "async": "2.1.5",
     "babel-polyfill": "6.23.0",
-    "browserify": "14.4.0",
+    "browserify": "16.2.0",
     "buffer": "5.0.6",
     "cli-table2": "0.2.0",
     "colors": "1.1.2",

--- a/src/utils/build.js
+++ b/src/utils/build.js
@@ -65,7 +65,7 @@ const requiredFiles = (cwd, entryPoints) => {
       'Buffer.isBuffer': undefined,
       Buffer: undefined
     },
-    ignoreMissing: false,
+    ignoreMissing: true,
     debug: false,
     standalone: undefined
   };
@@ -74,18 +74,11 @@ const requiredFiles = (cwd, entryPoints) => {
   return new Promise((resolve, reject) => {
     b.on('error', reject);
 
-    const ignoredDependencies = [
-      'debug' // Causes https://github.com/zapier/zapier-platform-cli/issues/189
-    ];
-
-    b.exclude(ignoredDependencies);
-
     const paths = [];
     b.pipeline.get('deps').push(
       through
         .obj((row, enc, next) => {
           const filePath = row.file || row.id;
-          // why does browserify add /private + filePath?
           paths.push(stripPath(cwd, filePath));
           next();
         })

--- a/src/utils/build.js
+++ b/src/utils/build.js
@@ -74,6 +74,12 @@ const requiredFiles = (cwd, entryPoints) => {
   return new Promise((resolve, reject) => {
     b.on('error', reject);
 
+    const ignoredDependencies = [
+      'debug' // Causes https://github.com/zapier/zapier-platform-cli/issues/189
+    ];
+
+    b.exclude(ignoredDependencies);
+
     const paths = [];
     b.pipeline.get('deps').push(
       through

